arch/arm/core/cortex_m/prep_c.c

```diff
@@ -39,22 +39,41 @@ Z_GENERIC_SECTION(.vt_pointer_section) __attribute__((used)) void *_vector_table
 
 #ifdef CONFIG_CPU_CORTEX_M_HAS_VTOR
 
-#define VECTOR_ADDRESS ((uintptr_t)_vector_start)
-
 /* In some Cortex-M3 implementations SCB_VTOR bit[29] is called the TBLBASE bit */
 #ifdef SCB_VTOR_TBLBASE_Msk
 #define VTOR_MASK (SCB_VTOR_TBLBASE_Msk | SCB_VTOR_TBLOFF_Msk)
 #else
 #define VTOR_MASK SCB_VTOR_TBLOFF_Msk
 #endif
 
+#ifdef CONFIG_SRAM_VECTOR_TABLE
+
+/* Vector table is in ROM, copy it to the RAM and set VTOR */
+
+#define VECTOR_ADDRESS ((uintptr_t)_sram_vector_start)
+
+void __weak relocate_vector_table(void)
+{
+	size_t vector_size = (size_t)_vector_end - (size_t)_vector_start;
+
+	memcpy(_sram_vector_start, _vector_start, vector_size);
+	SCB->VTOR = VECTOR_ADDRESS & VTOR_MASK;
+	barrier_dsync_fence_full();
+	barrier_isync_fence_full();
+}
```

Comment:
Do we really have to duplicate this routine?
Why not just do the following in the existing implementation?
```suggestion
#ifdef CONFIG_SRAM_VECTOR_TABLE
#define VECTOR_ADDRESS ((uintptr_t)_sram_vector_start)
#else
#define VECTOR_ADDRESS ((uintptr_t)_vector_start)
#endif

void __weak relocate_vector_table(void)
{

#ifdef CONFIG_SRAM_VECTOR_TABLE
	/* Copy vector table to its location in SRAM */
	size_t vector_size = (size_t)_vector_end - (size_t)_vector_start;

	memcpy(_sram_vector_start, _vector_start, vector_size);
#endif
	SCB->VTOR = VECTOR_ADDRESS & VTOR_MASK;
	barrier_dsync_fence_full();
	barrier_isync_fence_full();
}
```
(note: had to put the diff here because GitHub won't allow `suggestion` on untouched code ranges)